src/mmw/js/src/modeling/controllers.js

```diff
@@ -139,37 +133,31 @@ var ModelingController = {
                     lock.resolve();
                 }
 
-                lock.done(function() {
-                    project.on('change:id', updateUrl);
-                    initScenarioEvents(project);
-                    initViews(project);
-                    if (App.projectNumber) {
-                        updateUrl();
-                    }
-                });
+                finishProjectSetup(lock, project);
```

Comment:
In some functions `lock` is the first argument, and in others it is the second. Consider following a consistent convention.